nova/core/roles/os_configuration/tasks/routeros.yml

```diff
@@ -11,6 +11,21 @@
   community.routeros.command:
     commands: :exec {/system license generate-new-id}
 
-- name: Configuring licence...
-  community.routeros.command:
-    commands: /system license renew once level=p-unlimited account={{ routeros_license_account }} password={{ routeros_license_password }}
+- name: Configuring RouterOS licence...
+  when:
+    - routeros_license_account != {}
+    - routeros_license_password != {}
+  block:
+    - name: Configuring licence...
+      community.routeros.command:
+        commands: /system license renew once level=p-unlimited account={{ routeros_license_account }} password={{ routeros_license_password }}
+
+    - name: WARNING
+      ansible.builtin.debug:
+        msg: "RouterOS routeros_license_account or routeros_license_password is not defined, skipping license configuration"
+
+    - name: Sleeping to read WARNING...
+      ansible.builtin.wait_for:
+        timeout: 30
+      become: false
+      delegate_to: localhost
```

Comment:
Should not this two task be out of the block with a different when?

Reply:
Yes you are right. Fixed now